docs/cookbook/creating-webhooks.md

```diff
@@ -0,0 +1,105 @@
+          },
+          body: {
+            docTitle: data.published.title,
+            docId: data.published._id
+          }
+        };
+        const response = await self.apos.http.post(url, options);
+        // the response will depend on the endpoint you are sending to
+        // Potentially add code to handle errors coming back
+        // like retrying if the response isn't successful
+      }
+    }
+  };
+}
+```
+  <template v-slot:caption>
+    /modules/article/index.js
+    </template>
+
+</AposCodeBlock>
+
+In this block of code, we are using the `handlers(self)` module configuration function. We could use any server event, but in this case we are using the `afterPublication` event that is emitted by our custom article piece-type module. This server event delivers two parameters, `req` and `data`. The `data` parameter contains information about the document being published including the content and whether this is the first time it is being published. In this case, we are adding an early return if this isn't the first time the document is being published.
+
+The next section of code is involved in setting up the HTTP `POST` request and will depend heavily on what the desired endpoint requires, and what content you want sent to the endpoint. In this example, we are adding the piece `title` and `_id`.
+
+Finally, we are using the `post` method of the [`@apostrophecms/http` module](https://v3.docs.apostrophecms.org/reference/modules/http.html#async-post-url-options) to send our data to the endpoint. Depending on the endpoint, the returned response might be as simple as a `200` success or a `400` failed response, or might contain additional data. That response can be handled to retry a failed response or in some way log the returned data.
+
+## Setting up a webhook endpoint
+
+Your project may require a webhook endpoint for interfacing with a payment or subscription service. This can be accomplished by using the Apostrophe [`apiRoutes(self)` module configuration](https://v3.docs.apostrophecms.org/reference/module-api/module-overview.html#apiroutes-self) function. This allows you to easily set up an endpoint for any HTTP request method.
+
+<AposCodeBlock>
+
+```javascript
+options: {
+  csrfExceptions: [ '/webhooks' ]
+},
+apiRoutes(self) {
+  return {
+    post: {
+      // change function name to match the desired endpoint URL
+      '/webhooks': async function(req) {
+        
+        // Implement your own authorization check here
+        // to make sure the data is from a genuine source
+        // Each webhook provider likely has guidance and 
+        // a specific type of authorization
+        const authHeader = req.headers.authorization;
+        if (!authHeader) {
+          throw self.apos.error('forbidden');
+        }
+
+        // Use data passed by webhook to update database
+        // or provide some other type of functionality
+        const webhookData = req.body;
+        addNewSubscriber(webhookData);
```

Comment:
`await` this. The big win with apiRoutes is that they automatically handle exceptions for you and allow you to return values you got from async functions, but not if you don't await them. Then you just have a race condition.